R/transformCounts.R

```diff
@@ -317,6 +325,25 @@ setMethod("transformAssay", signature = c(x = "SummarizedExperiment"),
     return(mat)
 }
 
+####################################.apply_css###################################
+# This function applies cumulative sum scaling (CSS) to the abundance table.
+.calc_css <- function(mat, ...){
+  # If mat is a DelayedArray, convert it to a regular matrix for computation
+  if (inherits(mat, "DelayedArray")) {
+    mat <- as.matrix(mat)
+  }
+  
+  # Calculate the cumulative sum for each sample
+  cum_sum <- apply(mat, 2, cumsum)
```

Comment:
How about `colCumsums(mat)` so we can remove the step that converts DelayedMatrix to matrix?

Reply:
Add importFrom tag above .calc_css

Reply:
sorry as in how do you mean above .calc_css - I thought that kind of thing went in the namespace?

Reply:
Yes, they go, but namespace is generated automatically based on tags in R files. See other functions. IN your case, this should be something like this (syntax might be wrong)
```

#'  @importFrom MatrixGenerics colCumsum
.calc_css <- function(x, ...){

}
```

Reply:
ok thanks will do